cmd/template-deprecator/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowplugins "k8s.io/test-infra/prow/plugins"
+
+	"github.com/openshift/ci-tools/pkg/deprecatetemplates"
+)
+
+type options struct {
+	prowJobConfigDir     string
+	prowConfigPath       string
+	prowPluginConfigPath string
+	allowlistPath        string
+
+	help bool
+}
+
+func bindOptions(fs *flag.FlagSet) *options {
+	opt := &options{}
+
+	fs.StringVar(&opt.prowJobConfigDir, "prow-jobs-dir", "", "Path to a root of directory structure with Prow job config files (ci-operator/jobs in openshift/release)")
+	fs.StringVar(&opt.prowConfigPath, "prow-config-path", "", "Path to the Prow configuration file")
+	fs.StringVar(&opt.prowPluginConfigPath, "prow-plugin-config-path", "", "Path to the Prow plugin configuration file")
+	fs.StringVar(&opt.allowlistPath, "allowlist-path", "", "Path to template deprecation allowlist")
+
+	return opt
+}
+
+func (o *options) validate() error {
+	for param, value := range map[string]string{
```

Comment:
golly do i like anonymous maps

Reply:
Not sure if this is irony or not, but I think it's fine here :)